services/gitdiff/gitdiff.go

```diff
@@ -169,11 +169,15 @@ func getDiffLineSectionInfo(treePath, line string, lastLeftIdx, lastRightIdx int
 }
 
 // escape a line's content or return <br> needed for copy/paste purposes
-func getLineContent(content string) string {
+func getLineContent(content string) DiffInline {
 	if len(content) > 0 {
-		return html.EscapeString(content)
+		status, content := charset.EscapeControlString(content)
+		return DiffInline{
+			EscapeStatus: status,
+			Content:      template.HTML(html.EscapeString(content)),
```

Comment:
Hmm .... `EscapeControlString` generates `<span>`, and then will `EscapeString` escape these `<span>`?

Reply:
I tried to propose a PR https://github.com/zeripath/gitea/pull/10